stravalib/tests/integration/strava_api_stub.py

```diff
@@ -0,0 +1,109 @@
+import json
+import logging
+import os
+import re
+from functools import wraps, lru_cache
+from typing import Dict, Any, Callable, Optional
+
+import requests
+from responses import BaseResponse, RequestsMock
+
+from stravalib.protocol import ApiV3
+from stravalib.tests import RESOURCES_DIR
+
+LOGGER = logging.getLogger(__name__)
+
+
+@lru_cache(maxsize=1)
+def _get_strava_api_paths():
+    use_local = False
+
+    try:
+        strava_api_swagger_response = requests.get(
+            'https://developers.strava.com/swagger/swagger.json'
+        )
+        if strava_api_swagger_response.status_code != 200:
+            use_local = True
+    except requests.exceptions.ConnectionError:
+        use_local = True
+
+    if use_local:
+        LOGGER.warning(
+            'Failed to retrieve recent swagger API definition from Strava, using '
+            '(potentially stale) version from local resources'
+        )
+        with open(os.path.join(RESOURCES_DIR, 'strava_swagger.json'), 'r') as swagger_file:
+            return json.load(swagger_file)['paths']
+    else:
+        return strava_api_swagger_response.json()['paths']
+
+
+def _api_method_adapter(api_method: Callable) -> Callable:
+    @wraps(api_method)
+    def method_wrapper(
+            *args,
+            response_update: Dict[str, Any] = None,
+            n_results: Optional[int] = None,
+            **kwargs
+    ) -> BaseResponse:
```

Comment:
FYI, if we use type hint here we can not run these tests in python2.
But its fine, since we will drop python2 later

Reply:
if y'all are ok with dropping python 2.x i really think we should!

Reply:
I guess we agree that we'll not be supporting Python 2 any longer, so I'll keep the type hints.